flyteidl2/task/environment.proto

```diff
@@ -13,4 +13,7 @@ message Environment {
     (buf.validate.field).string.min_len = 1,
     (buf.validate.field).string.max_len = 63
   ];
+
+  // Optional description of environment
+  string description = 2;
```

Comment:
let's add the validation here also, we can always remove it in the future but i think safer to add for now.